### PR TITLE
First stab at rewriting SalesforceAnalytics tests with new testing framework

### PR DESCRIPTION
**Pull request by bhariharan**
`InstrumentationTestCase` has been deprecated. We should be using `InstrumentationRegistry` and the test support library instead - it's faster and more efficient.

### CODE REVIEW

File: libs/SalesforceAnalytics/build.gradle [hunk 1]
```diff
@@ -15,6 +15,8 @@ apply plugin: 'com.jfrog.bintray'
 
 dependencies {
     api 'com.squareup:tape:1.2.3'
+    androidTestApi 'com.android.support.test:runner:1.0.1'
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/build.gradle]: Adding support test libs for the test target.

File: libs/SalesforceAnalytics/build.gradle [hunk 2]
```diff
@@ -46,7 +48,7 @@ android {
   }
   defaultConfig {
     testApplicationId "com.salesforce.androidsdk.analytics.tests"
-    testInstrumentationRunner "com.salesforce.androidsdk.analytics.util.test.JUnitReportTestRunner"
+    testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/build.gradle]: We don't need our custom test runner anymore - it has been around for a while for legacy reasons. In the new framework you can specify annotations that define the size of the test and the framework will batch them accordingly.

File: libs/test/SalesforceAnalyticsTest/AndroidManifest.xml [hunk 3]
```diff
@@ -9,9 +9,6 @@
 		<uses-library android:name="android.test.runner" />
 	</application>
 
-	<instrumentation android:targetPackage="com.salesforce.androidsdk.analytics.tests"
```
Comment (bhariharan) [inline libs/test/SalesforceAnalyticsTest/AndroidManifest.xml]: We don't need this anymore.

Comment (bhariharan) [inline libs/test/SalesforceAnalyticsTest/AndroidManifest.xml]: We don't need this anymore.

File: libs/test/SalesforceAnalyticsTest/src/com/salesforce/androidsdk/analytics/logger/FileLoggerTest.java [hunk 4]
```diff
@@ -37,7 +46,9 @@
  *
  * @author bhariharan
  */
-public class FileLoggerTest extends InstrumentationTestCase {
+@RunWith(AndroidJUnit4.class)
```
Comment (bhariharan) [inline libs/test/SalesforceAnalyticsTest/src/com/salesforce/androidsdk/analytics/logger/FileLoggerTest.java]: This annotation uses `JUnit4` - `JUnit3` is the other option.

File: libs/test/SalesforceAnalyticsTest/src/com/salesforce/androidsdk/analytics/logger/FileLoggerTest.java [hunk 5]
```diff
@@ -37,7 +46,9 @@
...
+@SmallTest
```
Comment (bhariharan) [inline libs/test/SalesforceAnalyticsTest/src/com/salesforce/androidsdk/analytics/logger/FileLoggerTest.java]: This specifies the size of the tests in this class.

File: libs/test/SalesforceAnalyticsTest/src/com/salesforce/androidsdk/analytics/logger/FileLoggerTest.java [hunk 6]
```diff
@@ -49,22 +60,20 @@
     private Context targetContext;
     private FileLogger fileLogger;
 
-    @Override
+    @Before
```
Comment (bhariharan) [inline libs/test/SalesforceAnalyticsTest/src/com/salesforce/androidsdk/analytics/logger/FileLoggerTest.java]: No more extending/sub-classing - everything is driven by annotations.

File: libs/test/SalesforceAnalyticsTest/src/com/salesforce/androidsdk/analytics/logger/FileLoggerTest.java [hunk 7]
```diff
@@ -74,247 +83,262 @@ public void tearDown() throws Exception {
...
-        assertEquals("Max size didn't match expected max size", newMaxSize, maxSize);
+        Assert.assertEquals("Max size didn't match expected max size", newMaxSize, maxSize);
     }
 
     /**
      * Test for flushing the log file.
      *
      * @throws Exception
      */
+    @Test
```
Comment (bhariharan) [inline libs/test/SalesforceAnalyticsTest/src/com/salesforce/androidsdk/analytics/logger/FileLoggerTest.java]: Need this annotation to make the test run.

**Comment by brandonpage** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_ci/circleci">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**